Common/Securities/Volatility/RelativeStandardDeviationVolatilityModel.cs

```diff
@@ -109,25 +113,37 @@ public void Update(Security security, BaseData data)
         /// <param name="security">The security of the request</param>
         /// <param name="utcTime">The date/time of the request</param>
         /// <returns>History request object list, or empty if no requirements</returns>
-        public IEnumerable<HistoryRequest> GetHistoryRequirements(Security security, DateTime utcTime)
+        public override IEnumerable<HistoryRequest> GetHistoryRequirements(Security security, DateTime utcTime)
         {
             var barCount = _window.Size + 1;
             var localStartTime = Time.GetStartTimeForTradeBars(security.Exchange.Hours, utcTime.ConvertFromUtc(security.Exchange.TimeZone), _periodSpan, barCount, security.IsExtendedMarketHours);
             var utcStartTime = localStartTime.ConvertToUtc(security.Exchange.TimeZone);
 
-            return new[] 
+            if (SubscriptionDataConfigProvider == null)
             {
-                new HistoryRequest(utcStartTime, 
+                throw new Exception(
+                    "RelativeStandardDeviationVolatilityModel.GetHistoryRequirements(): " +
+                    "SubscriptionDataConfigProvider was not set."
+                );
+            }
+            var configurations = SubscriptionDataConfigProvider
+                .GetSubscriptionDataConfigs(security.Symbol)
+                .ToList();
+            var configuration = configurations.First();
```

Comment:
True, in this case, the first SDC is used to obtain `.Symbol` and `.DataTimeZone`. Primarily the objective was to avoid doing `MarketHoursDatabase.FromDataFolder().GetDataTimeZone()` which did not seem to be required given the data was in context.

Reply:
hmm, interesting dilemma indeed. The system as written actually _assumes_ all of a security's data have the same market hours entries (by virtue of the market-hours-db and it's consumers) -- so if that's the only value being pulled from the config than I'd say that's completely safe 👍